### ISSUE TITLE
firebase deploy issues

### ISSUE THREAD

**Issue by jnash766** [mdanics/fluttergram#96]
I'm very new to firebase and trying to work through the setup. I ran into an issue I can't seem to get past. I apologize in advance, I'm sure this must be something dumb I'm missing.

When doing this step:
Push the function getFeed with firebase deploy --only functions In the output, you'll see the getFeed URL, copy that.

I get the following error:
John-iMac lib % firebase deploy --only functions
⚠  functions: package.json indicates an outdated version of firebase-functions.
Please upgrade using npm install --save firebase-functions@latest in your functions directory.

=== Deploying to 'fluttergram-xxxxx'...

i  deploying functions
Running command: npm --prefix "$RESOURCE_DIR" run lint

> functions@ lint /Users/John/Desktop/Coding/Flutter/fluttergram/functions/functions
> tslint --project tsconfig.json

Running command: npm --prefix "$RESOURCE_DIR" run build

> functions@ build /Users/John/Desktop/Coding/Flutter/fluttergram/functions/functions
> tsc

src/index.ts:1:1 - error TS6133: 'functions' is declared but its value is never read.

1 import * as functions from 'firebase-functions';
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! functions@ build: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the functions@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

**Comment by jnash766** [comment]
I tried going through the all the steps again, including a new clone of the project.  Still seem to be stuck, but guessing I'm missing something obvious.
Thanks!


John@John-iMac fluttergram % ls
CHANGELOG.md			android.iml			firestore_test_android.iml	pubspec.lock
LICENSE				app-release.apk			functions			pubspec.yaml
README.md			assets				ios
android				firestore_test.iml		lib
John@John-iMac fluttergram % cd functions
John@John-iMac functions % ls
index.js		package-lock.json	src			tslint.json
lib			package.json		tsconfig.json
John@John-iMac functions % firebase init

     ######## #### ########  ######## ########     ###     ######  ########
     ##        ##  ##     ## ##       ##     ##  ##   ##  ##       ##
     ######    ##  ########  ######   ########  #########  ######  ######
     ##        ##  ##    ##  ##       ##     ## ##     ##       ## ##
     ##       #### ##     ## ######## ########  ##     ##  ######  ########

You're about to initialize a Firebase project in this directory:

  /Users/John/Desktop/Coding/fluttergram/functions

? Which Firebase CLI features do you want to set up for this folder? Press Space to select features, then Enter to confirm your choices. Functions: Configu
re and deploy Cloud Functions

=== Project Setup

First, let's associate this project directory with a Firebase project.
You can create multiple project aliases by running firebase use --add, 
but for now we'll just set up a default project.

? Please select an option: Use an existing project
? Select a default Firebase project for this directory: fluttergram-d6c65 (fluttergram)
i  Using project fluttergram-d6c65 (fluttergram)

=== Functions Setup

A functions directory will be created in your project with a Node.js
package pre-configured. Functions can be deployed with firebase deploy.

? What language would you like to use to write Cloud Functions? TypeScript
? Do you want to use TSLint to catch probable bugs and enforce style? Yes
✔  Wrote functions/package.json
✔  Wrote functions/tslint.json
✔  Wrote functions/tsconfig.json
✔  Wrote functions/src/index.ts
✔  Wrote functions/.gitignore
? Do you want to install dependencies with npm now? Yes

> protobufjs@6.9.0 postinstall /Users/John/Desktop/Coding/fluttergram/functions/functions/node_modules/protobufjs
> node scripts/postinstall

npm notice created a lockfile as package-lock.json. You should commit this file.
added 287 packages from 221 contributors and audited 287 packages in 22.754s

31 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities


i  Writing configuration info to firebase.json...
i  Writing project information to .firebaserc...
i  Writing gitignore file to .gitignore...

✔  Firebase initialization complete!
John@John-iMac functions % firebase deploy

=== Deploying to 'fluttergram-d6c65'...

i  deploying functions
Running command: npm --prefix "$RESOURCE_DIR" run lint

> functions@ lint /Users/John/Desktop/Coding/fluttergram/functions/functions
> tslint --project tsconfig.json

Running command: npm --prefix "$RESOURCE_DIR" run build

> functions@ build /Users/John/Desktop/Coding/fluttergram/functions/functions
> tsc

src/index.ts:1:1 - error TS6133: 'functions' is declared but its value is never read.

1 import * as functions from 'firebase-functions';
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! functions@ build: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the functions@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/John/.npm/_logs/2020-06-30T20_05_19_692Z-debug.log

Error: functions predeploy error: Command terminated with non-zero exit code2
John@John-iMac functions % ls
firebase.json		index.js		package-lock.json	src			tslint.json
functions		lib			package.json		tsconfig.json
John@John-iMac functions %                
John@John-iMac functions % 
John@John-iMac functions % 
John@John-iMac functions % 
John@John-iMac functions % 
John@John-iMac functions % firebase deploy --only functions

=== Deploying to 'fluttergram-d6c65'...

i  deploying functions
Running command: npm --prefix "$RESOURCE_DIR" run lint

> functions@ lint /Users/John/Desktop/Coding/fluttergram/functions/functions
> tslint --project tsconfig.json

Running command: npm --prefix "$RESOURCE_DIR" run build

> functions@ build /Users/John/Desktop/Coding/fluttergram/functions/functions
> tsc

src/index.ts:1:1 - error TS6133: 'functions' is declared but its value is never read.

1 import * as functions from 'firebase-functions';
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! functions@ build: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the functions@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/John/.npm/_logs/2020-06-30T20_06_44_411Z-debug.log

Error: functions predeploy error: Command terminated with non-zero exit code2

Having trouble? Try firebase [command] --help

**Comment by jnash766** [comment]
Not sure if this is ok, but I found where other firebase projects were having this same error.  They were able to resolve it by editing the tsconfig.json and changing the following:

"noUnusedLocals": true,    <-- change this to: false

I did this and I no longer get an error, but I also don't see "getFeed URL" in the output.  Any ideas if thats ok to change that variable and if where I would find the getFeed URL?  I'm guessing the project console URL is not what I'm looking for?


John@John-iMac functions % firebase deploy --only functions

=== Deploying to 'fluttergram-xxxx'...

i  deploying functions
Running command: npm --prefix "$RESOURCE_DIR" run lint

> functions@ lint /Users/John/Desktop/Coding/fluttergram/functions/functions
> tslint --project tsconfig.json

Running command: npm --prefix "$RESOURCE_DIR" run build

> functions@ build /Users/John/Desktop/Coding/fluttergram/functions/functions
> tsc

✔  functions: Finished running predeploy script.
i  functions: ensuring required API cloudfunctions.googleapis.com is enabled...
✔  functions: required API cloudfunctions.googleapis.com is enabled
i  functions: preparing functions directory for uploading...

✔  Deploy complete!

Project Console: https://console.firebase.google.com/project/fluttergram-xxxx/overview
John@John-iMac functions %
